service/state/core_access.go

```diff
@@ -66,6 +68,15 @@ func (ca *CoreAccessor) Stop(context.Context) error {
 	return nil
 }
 
+func (ca *CoreAccessor) SubmitPayForData(
+	ctx context.Context,
+	nID namespace.ID,
+	data []byte,
+	gasLim uint64,
```

Comment:
Q: why do you need a gas limit as a parameter her? The gas cost of PFD txs is known from the size of the tx in bytes.
cc @evan-forbes

Reply:
True! Using the current gas pricing, we could hardcore this. Gas price however we might want to expose.

Reply:
can we hardcode gas lim in the app?

Reply:
yes, we could, and we might want to swap `gasLim` with `gasPrice` here, or alternatively leave it blank and rely on [passing options](https://github.com/celestiaorg/celestia-app/blob/d7bb3934e0c13259bf3d666378ea67fda852b6e5/x/payment/payfordata.go#L30)

Reply:
ahhh wait, I think I remember trying this ages ago, and unless we want to hardcode an overly large limit, when submitting the wirePFD, CheckTx will sometimes return an error even though it should and the tx doesn't make it to the mempool :disappointed: 

are we okay with hardcoding a large value, at least for now?

Reply:
https://github.com/celestiaorg/celestia-app/issues/378

Reply:
@evan-forbes, can you pls clarify how decision in https://github.com/celestiaorg/celestia-app/issues/378 affects the `gasLim` param here?

Reply:
yeah, the gasLim needs to be here, as the gas limit will change depending on the size of the data being posted. Previously we discussed that the gas limit should be identical for all PFDs, but that's not the case, as the WirePFD (which includes the data) is checked before being admitted to the mempool, and we will deduct the appropriate amount of gas in the state machine soon.

Reply:
can we resolve this? @evan-forbes @Wondertan ?